vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -67,6 +77,39 @@
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
 
+  - name: Build IPAM manifests
+    ansible.builtin.command: "{{ item }}"
+    args:
+      chdir: "{{ CAPM3PATH }}"
+    loop:
+      - make hack/tools/bin/kustomize
+      - ./hack/tools/bin/kustomize build "{{ IPAMPATH }}/config/default" --output config/ipam/metal3-ipam-components.yaml
+
+  # TODO: update capm3 images (before making manifests)
+  # If we want to use the local registry and/or proxy we must make changes to the
+  # kustomizations before making the manifests.
```

Comment:
Note this!
I'm not sure what to do about the images. Previously, when we just updated a patch version, I think the same image was reused (so not really an upgrade to the images, just to the CRDs). If the images actually were updated we used the upstream directly (not the local registry or proxy).

In 03_launch_management_cluster.sh we change the images [like this](https://github.com/metal3-io/metal3-dev-env/blob/620b4e6abef8c0edaedd8dc3cdab2a5b3d0da7bd/03_launch_mgmt_cluster.sh#L324-L351). But we do not have any variables for what images to use after the upgrade so it would take a bit of work to set it up. Don't get me wrong, I think it would be good to do it, I'm just not sure if it is worth to do at this point. The upgrade tests are not run that often so rate limits may not be that important. What do you think?

Reply:
I'm not that much familiar with upgrade tests, but I believe you are correct that we were re-using the same image.

Reply:
I have now added variables for the upgraded images and made the changes required to use them. It does not use the local registry but it does respect the `CONTAINER_REGISTRY` variable.